scipy/stats/tests/test_distributions.py

```diff
@@ -6934,14 +6934,15 @@ def test_cdf(self):
         assert_allclose(p[1], 1 - np.arctan(cr*np.tan(np.pi - x2/2))/np.pi)
 
 
-def test_rvs_no_size_warning():
+def test_rvs_no_size_error():
+    # _rvs methods must have parameter `size`; see gh-11394
     class rvs_no_size_gen(stats.rv_continuous):
         def _rvs(self):
             return 1
 
     rvs_no_size = rvs_no_size_gen(name='rvs_no_size')
 
-    with assert_warns(np.VisibleDeprecationWarning):
+    with assert_raises(TypeError, match=re.escape("_rvs() got an unexpected")):
```

Comment:
Prefer usage of raw strings. There should be no need for `re`
```suggestion
    with assert_raises(TypeError, match=r"_rvs() got an unexpected"):
```

Reply:
I do. I tried it because of the last time you suggested it. But it doesn't work. I just tried again:
```
AssertionError: Regex pattern '_rvs() got an unexpected' does not match "_rvs() got an unexpected keyword argument 'size'".
```

Reply:
Oh, I see. Thanks for the precisions. Then sure. I will play later locally with that. I thought I used similar things in the past.